openhcl/virt_mshv_vtl/src/processor/mshv/arm64.rs

```diff
@@ -217,9 +233,8 @@ impl BackingPrivate for HypervisorBackedArm64 {
     /// The VTL that was running when the VP exited into VTL2, with the
     /// exception of a successful vtl switch, where it will return the VTL
     /// that will run on VTL 2 exit.
-    fn last_vtl(_this: &UhProcessor<'_, Self>) -> Vtl {
-        // TODO ARM64
-        Vtl::Vtl0
+    fn last_vtl(this: &UhProcessor<'_, Self>) -> Vtl {
+        this.backing.last_vtl.unwrap_or(Vtl::Vtl0)
```

Comment:
Once we get this from the register page, we should not have an unwrap path. Ideally this just reads directly from the register page and doesn't copy anything locally.

Reply:
My understanding is that the register page is only valid on the intercept path. So in general, I was thinking that if the register page doesn't exist, calls to last_vtl should panic, but I was also concerned about breaking things with that approach. Do you have concerns with panicking if the register page doesn't exist?

Reply:
The register page is supposed to always be valid if we're running in a higher VTL. But if we can't use a register page, then we can't just assume VTL0 and hope for the best. We have to statically ensure that we never need the VTL outside of the intercept handling path. That probably means removing `last_vtl()` from the trait and instead pushing it down the function call graph as necessary. E.g., add it as a parameter to `write_msr`.

I don't like that at all. I'd much rather get the required support from the hypervisor. If they don't want to give us a register page, can they put the last VTL on the VP assist page?

Reply:
I can ask if they can put the last VTL on the VP assist page, but for full correctness, wouldn't we still need to be able to handle downlevel hypervisors that don't put the last VTL on the VP assist page?

Reply:
I think we can say that we only support VSM on newer hypervisors.

Reply:
The feedback I'm getting is that the "last vtl" is only valid when VTL 2 is handling an intercept, and is unreliable otherwise. e.g. if VTL 0 makes a VTL call at the same time that VTL 2 receives an interrupt, then whatever last VTL the hypervisor reports to VTL 2 is a matter of timing.

Reply:
Since last vtl in general is only valid when VTL 2 is handling an intercept, I renamed it to "intercepted vtl" and removed the last_vtl trait. I tried to reduce the amount of function signature changes.